system/modules/rest/models/SearchableService.php

```diff
@@ -0,0 +1,288 @@
+<?php
+/**
+	 * Service providing search function where search is configured by an array of parameters that match the REST style URL
+	 * eg /SKIP/10/LIMIT/10/AND/name___like/fred/age___between/40___60
+	 * eg /SKIP/10/LIMIT/10/AND/name___like/fred/OR/age__between/0___20/age___greater/80
+	 * search config can start with 
+	 * 	- SKIP
+	 * 	- LIMIT
+	 * 	- OR
+	 * 	- AND
+	 * 	- <operator>
+	 * subsequent search config is processed as follows
+	 * - SKIP and LIMIT are extracted for query generation
+	 * - if AND or OR is found, a query group is created
+	 * 		- END closes as sub group
+	 * - otherwise config pairs are processed as <field__operator>, <data1__data2> until AND or OR is found or the end of the configuration
+	 * 		- if AND or OR is found again, a sub query group is created
+	 * @param Boolean $useCache
+	 * 
+	 * @author Steve Ryan
+ */
+class QueryPart {
+	public $filter='';
+	public $parameters=[];
+}
+ 
+class SearchableService extends DbService {
```

Comment:
Can we prefix this to 'RestSearchableService'? As we don't use namespaces I fear that this could easily conflict with our search module.